pandas/core/frame.py

```diff
@@ -3386,12 +3387,8 @@ class    max    type
             new_obj = self.copy()
 
         def _maybe_casted_values(index, labels=None):
-            if isinstance(index, PeriodIndex):
-                values = index.astype(object).values
-            elif isinstance(index, DatetimeIndex) and index.tz is not None:
-                values = index
-            else:
-                values = index.values
+            values = index._values
```

Comment:
hmm I'm not really sure :) It also seems to just be used in one place, so this seems like an opportunity for cleanup.

This change is possible because `PeriodIndex.values / _values` is just `.astype(object).values`, and `DatetimeIndex._values` is `self` with a TZ.